### PR TITLE
Add `system-characteristics` `has-data-flow` constraints

### PR DESCRIPTION
**Pull request by Gabeblis**
# Committer Notes
### Description
Added `system-characteristics` `has-data-flow` constraints and tests for each of the constraints. These should be added as they are part of the effort write all of the constraints from the constraint tracker.

### Changes
- Added constraints `has-data-flow`, `has-data-flow-diagram`, `has-data-flow-description`, `has-data-flow-diagram-uuid`, `has-data-flow-diagram-description`, `has-data-flow-diagram-link`, `has-data-flow-diagram-caption`, `has-data-flow-diagram-link-rel`, and `has-data-flow-diagram-link-rel-allowed-value` to fedramp-external-constraints.xml.
- Added pass and fail yaml tests for all of the above constraints.
- Edited `ssp-all-VALID.xml` to ensure all constraints pass correctly.
- Added `-INVALID.xml` file for each constraint.

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
~~- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?~~
~~- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?~~

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 1]
```diff
@@ -110,6 +110,36 @@
...
+            </expect>
+            <expect id="has-data-flow-diagram" target="system-characteristics/data-flow" test="diagram" level="ERROR">
+                <message>A FedRAMP SSP must have at least one data flow diagram.</message>
+            </expect>
+            <expect id="has-data-flow-description" target="system-characteristics/data-flow" test="description" level="ERROR">
+                <message>An OSCAL SSP document with a data flow must have a description.</message>
+            </expect>
+            <expect id="has-data-flow-description" target="system-characteristics/data-flow" test="description" level="ERROR">
+                <message>An OSCAL SSP document with a data flow must have a description.</message>
+            </expect>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:124]: Blocking: dupe?

```suggestion
```

File: src/validations/constraints/fedramp-external-constraints.xml [hunk 2]
```diff
@@ -110,6 +110,36 @@
             <expect id="has-network-architecture-diagram-link-rel-allowed-value" target="system-characteristics/network-architecture/diagram/link" test="@rel eq 'diagram'" level="ERROR">
                 <message>Each FedRAMP SSP network architecture diagram must have a link rel attribute with the value "diagram".</message>
             </expect>
+            <expect id="has-data-flow" target="system-characteristics" test="data-flow" level="WARNING">
+                <message>A FedRAMP SSP must include a data flow diagram.</message>
+            </expect>
```
Comment (aj-stein-gsa) [inline src/validations/constraints/fedramp-external-constraints.xml:115]: Minor, non-blocking: given the next one about diagrams with the `<data-flow/>` assembly, can we make the message less duplicative here?

```suggestion
            <expect id="has-data-flow" target="system-characteristics" test="data-flow" level="WARNING">
                <message>A FedRAMP SSP must include a data flow section.</message>
            </expect>
```

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 1 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: Resolved, thanks :)

File: src/validations/constraints/fedramp-external-constraints.xml (hunk 2 above)
Comment (Gabeblis) [reply src/validations/constraints/fedramp-external-constraints.xml]: Good point, I added your suggestion.

**Review by aj-stein-gsa — APPROVED** [review]

**Pull request merged**